src/zepben/evolve/services/network/tracing/connectivity/connectivity_trace.py

```diff
@@ -5,32 +5,46 @@
 #  file, You can obtain one at https://mozilla.org/MPL/2.0/.
 from __future__ import annotations
 
-import itertools
 from typing import TYPE_CHECKING, TypeVar
+from zepben.evolve import BusbarSection, Queue,  BasicTraversal, ConnectivityTracker, connected_terminals
 
-from zepben.evolve.services.network.network_service import connected_terminals
 if TYPE_CHECKING:
-    from zepben.evolve import ConnectivityResult, BusbarSection, Traversal
+    from zepben.evolve import ConnectivityResult
     from zepben.evolve.types import OpenTest, QueueNext
     T = TypeVar("T")
 
-__all__ = ["queue_next_connectivity_result_with_open_test"]
+__all__ = ["create_connectivity_traversal"]
 
 
-# TODO rename this to something sane
-def queue_next_connectivity_result_with_open_test(open_test: OpenTest) -> QueueNext[ConnectivityResult]:
-    def queue_next(cr: ConnectivityResult, traversal: Traversal[ConnectivityResult]):
+def create_connectivity_traversal(open_test: OpenTest, queue: Queue[ConnectivityResult]):
```

Comment:
In Python, default arguments are evaluated at definition, not for each execution. Mutable default arguments (e.g. lists) should instead use `None` as a stand-in to be replaced in the function body, lest each additional call produce a reference to the same mutable object: https://pythonconquerstheuniverse.wordpress.com/2012/02/15/mutable-default-arguments/

I added a test to ensure this in `test_connectivity_trace.py`. I also ended up translating tests for `BasicTraversal` because I wanted to test that class field values are not copied to instances (`process_queue` and `tracker` in particular). I found that dataclassy takes care of that by turning them into factories, but fixed an issue where not all stop conditions would be checked due to short circuiting in the process.

Reply:
TL;DR: Used `= None` to accomplish this, wrote tests for `BasicTraversal`, fixed short-circuiting when checking stop conditions.